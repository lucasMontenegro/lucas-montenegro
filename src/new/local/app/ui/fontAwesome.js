const { library } = require("@fortawesome/fontawesome-svg-core")
const { fab } = require("@fortawesome/free-brands-svg-icons")
const { fas } = require("@fortawesome/free-solid-svg-icons")
const { far } = require("@fortawesome/free-regular-svg-icons")
library.add(fab, fas, far)