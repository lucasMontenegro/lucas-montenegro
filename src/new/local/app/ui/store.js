import React from "react"
import { createStore, combineReducers } from "redux"
import { Provider } from "react-redux"
import Routes from "new/local/app/ui/Routes"
export const reducers = {
  core (state={}, action) {
    return state
  },
}
export default (
  <Provider store={createStore(combineReducers(reducers))}>
    <Routes />
  </Provider>
)