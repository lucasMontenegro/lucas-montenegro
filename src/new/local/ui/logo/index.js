import React from "react"
import { ReactComponent as Logo } from "./logo.svg"
export default (<Logo width="32" height="32" aria-label="Lucas Montenegro logo" />)