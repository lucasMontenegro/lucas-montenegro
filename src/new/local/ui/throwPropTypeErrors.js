import globals from "new/local/utils/globals"
import isProduction from "new/local/utils/isProduction"
export default function throwPropTypeErrors () {
  if (isProduction()) {
    return
  }
  const { error } = globals.console
  globals.console.error = function logErrorOrThrow (msg) {
    if (/invalid|failed/gi.test(msg) && /prop.?type/gi.test(msg)) {
      throw new Error(msg)
    }
    error.apply(this, arguments)
  }
}