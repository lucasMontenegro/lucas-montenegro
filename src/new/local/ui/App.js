import createApp from "new/local/paperbase/createApp"
import makeTranslations from "new/local/utils/makeTranslations"
import routing from "new/local/ui/routing"
import logo from "new/local/ui/logo"
import home from "new/local/home"
import notFound from "new/local/notFound"
export default createApp({
  name: `main`,
  routing,
  logo,
  clients: {
    home: home(routing),
    notFound: notFound(routing),
  },
  titles: makeTranslations(() => `Lucas Montenegro`),
})