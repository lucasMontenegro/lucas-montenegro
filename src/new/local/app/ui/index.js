import ReactDOM from "react-dom"
import styling from "new/local/app/ui/styling"
import throwPropTypeErrors from "new/local/app/ui/throwPropTypeErrors"
import globals from "new/local/utils/globals"
throwPropTypeErrors()
ReactDOM.render(styling, globals.document.getElementById(`root`))