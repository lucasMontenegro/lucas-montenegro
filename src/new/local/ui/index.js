import ReactDOM from "react-dom"
import styling from "new/local/ui/styling"
import throwPropTypeErrors from "new/local/ui/throwPropTypeErrors"
import globals from "new/local/utils/globals"
throwPropTypeErrors()
ReactDOM.render(styling, globals.document.getElementById(`root`))