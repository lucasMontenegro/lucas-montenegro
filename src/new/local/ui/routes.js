import React from "react"
import { BrowserRouter, Route } from "react-router-dom"
import App from "new/local/ui/App"
export default (
  <BrowserRouter>
    <Route component={App} />
  </BrowserRouter>
)