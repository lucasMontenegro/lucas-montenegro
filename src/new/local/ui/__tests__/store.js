jest.mock(`new/local/ui/Routes`, () => ({
  __esModule: true,
  default: () => `new/local/ui/Routes`,
}))
describe(`new/local/ui/store`, () => {
  it(`should utilize the dependency APIs correctly`, () => {
    expect(jestUtils.getDependencies([`react`, `redux`, `react-redux`])).toMatchSnapshot()
  })
  it(`should run`, () => {
    require("new/local/ui/store")
  })
  describe(`core reducer`, () => {
    it(`should run`, () => {
      require("new/local/ui/store").reducers.core()
    })
  })
})