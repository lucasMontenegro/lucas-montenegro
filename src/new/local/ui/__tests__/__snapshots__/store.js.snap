// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new/local/ui/store should utilize the dependency APIs correctly 1`] = `
Object {
  "react": "^16.10.1",
  "react-redux": "^7.1.1",
  "redux": "^4.0.4",
}
`;
